plugins/soundsourcemediafoundation/soundsourcemediafoundation.cpp

```diff
@@ -160,233 +124,273 @@ void SoundSourceMediaFoundation::close() {
 SINT SoundSourceMediaFoundation::seekSampleFrame(
         SINT frameIndex) {
     DEBUG_ASSERT(isValidFrameIndex(frameIndex));
-    if (sDebug) {
-        qDebug() << "seekSampleFrame()" << frameIndex;
-    }
-    qint64 mfSeekTarget(mfFromFrame(frameIndex, getSamplingRate()) - 1);
-    // minus 1 here seems to make our seeking work properly, otherwise we will
-    // (more often than not, maybe always) seek a bit too far (although not
-    // enough for our calculatedFrameFromMF <= nextFrame assertion in ::read).
-    // Has something to do with 100ns MF units being much smaller than most
-    // frame offsets (in seconds) -bkgood
-    SINT result = m_iCurrentPosition;
-    if (m_dead) {
-        return result;
-    }
 
-    PROPVARIANT prop;
-    HRESULT hr;
+    if (m_currentFrameIndex < frameIndex) {
+        // seeking forward
+        SINT skipFramesCount = frameIndex - m_currentFrameIndex;
+        if (skipFramesCount <= kNumberOfPrefetchFrames) {
+            skipSampleFrames(skipFramesCount);
+            DEBUG_ASSERT(m_currentFrameIndex == frameIndex);
+        }
+    }
 
-    // this doesn't fail, see MS's implementation
-    hr = InitPropVariantFromInt64(mfSeekTarget < 0 ? 0 : mfSeekTarget, &prop);
+    if (m_currentFrameIndex == frameIndex) {
+        // already there
+        return m_currentFrameIndex;
+    }
 
-    hr = m_pReader->Flush(MF_SOURCE_READER_FIRST_AUDIO_STREAM);
-    if (FAILED(hr)) {
-        qWarning() << "SSMF: failed to flush before seek";
+    if (m_pSourceReader == nullptr) {
+        // reader is dead -> jump to end of stream
+        return getMaxFrameIndex();
     }
 
-    // http://msdn.microsoft.com/en-us/library/dd374668(v=VS.85).aspx
-    hr = m_pReader->SetCurrentPosition(GUID_NULL, prop);
+    HRESULT hr;
+
+    hr = m_pSourceReader->Flush(MF_SOURCE_READER_FIRST_AUDIO_STREAM);
```

Comment:
Here a comment can be handy, please improve: 
// Here we need to either jump far ahead of backward so the stored frames in th reader a useless.